### PR TITLE
add patch to make ncbi-vdb 3.0.0 compatible with HDF5 1.12.2

### PR DESCRIPTION
**Pull request by boegel**
follow-up for #16376, with improvements that were implemented as a part of #16908

### CODE REVIEW

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Comment by boegel** [comment]
@boegelbot please test @ jsc-zen2

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3155.skitty.os - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/226162c9b0f0d6ff1f8914a035454f4f for a full test report.

File: easybuild/easyconfigs/n/ncbi-vdb/ncbi-vdb-3.0.0-gompi-2022a.eb [hunk 1]
```diff
@@ -29,29 +31,17 @@ builddependencies = [
...
 
 dependencies = [
-    ('file', '5.43'),  # provides libmagic
-    ('HDF5', '1.10.8'),  # version 1.12.x has changes to API and is not compatible
+    ('HDF5', '1.12.2'),
     ('libxml2', '2.9.13'),
     ('bzip2', '1.0.8'),
 ]
 
-# add addtional libraries needed to statically link HDF5 from EB
```
Comment (verdurin) [inline easybuild/easyconfigs/n/ncbi-vdb/ncbi-vdb-3.0.0-gompi-2022a.eb:38]: @boegel are these settings for static linking really no longer required?

**Review by verdurin — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/n/ncbi-vdb/ncbi-vdb-3.0.0-gompi-2022a.eb (hunk 1 above)
Comment (boegel) [reply easybuild/easyconfigs/n/ncbi-vdb/ncbi-vdb-3.0.0-gompi-2022a.eb:38]: The patching of `konfigure.perl` is only relevant when using the provided `configure` script (which calls `konfigure.perl`), but we're bypassing that and calling `cmake` ourselves here - see also https://github.com/ncbi/ncbi-vdb/wiki/Building-and-Installing-from-Source

**Comment by verdurin** [comment]
Test report by @verdurin
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
easybuild-c7.novalocal - Linux CentOS Linux 7.9.2009, x86_64, Intel Xeon Processor (Skylake, IBRS), Python 3.6.8
See https://gist.github.com/5461d057b46987998ccfa7f9002b97a2 for a full test report.

**Review by verdurin — APPROVED** [review]
Looks fine.

**Comment by verdurin** [comment]
Going in, thanks @boegel!

**Pull request merged**